applepay/utils.py

```diff
@@ -66,3 +100,316 @@ def signing_time_is_valid(signature, current_time, threshold):
+
+    class_, method, and tag are needed to emit the BER
+    encoded version of the header + content.
+    These values are not exposed on SignerInfo
+    so we get them from the parent class.
+
+    Args:
+        signed_attrs (asn1crypto.cms.CMSAttributes): The signed
+            attributes from the signature
+    Returns:
+        string: The BER-encoded signed attributes as a string
+            of bytes
+    """
+    class_ = super(signed_attrs.__class__, signed_attrs).class_
+    method = super(signed_attrs.__class__, signed_attrs).method
+    tag = super(signed_attrs.__class__, signed_attrs).tag
+
+    return emit(class_, method, tag, signed_attrs.contents)
+
+
+def remove_ec_point_prefix(point):
+    """Remove the prefix from the uncompressed ec point.
+
+    We expect the point to be in the uncompressed
+    format described here: https://tools.ietf.org/html/rfc5480#section-2.2
+    This means the first byte must be "\x04" otherwise consider
+    the public key not usable.  ecdsa.keys.VerifyingKey does not
+    handle this first byte so its chopped off before turning the public
+    key bytes.
+
+    Args:
+        point (string): The uncompressed byte string of an EC Point
+    Returns:
+        string: The EC point byte string minus the uncompressed
+            byte prefix indicator
+    """
+    if not point.startswith("\x04"):
+        logger.warning("Expected uncompressed EC point.")
+        return None
+
+    return point[1:]
+
+
+def get_first_from_iterable(filter_func, iterable):
+    """Get the first filtered item from an iterable.
+
+    Args:
+        filter_func (callable): the function to filter on
+        iterable (iterable): the iterable to filter on
+    Returns:
+        object: the first filtered item from iterable or None
+            if no items matching the filter are found
+    """
+    filtered_stuff = ifilter(filter_func, iterable)
+    return next(filtered_stuff, None)
+
+
+def get_hashfunc_by_name(name, data):
+    """
+    Get a callable hashfunc by name.
+
+    This function can be used directly or with functools.partial, for example:
+
+    >>> hashfunc = functools.partial(get_hashfunc_by_name, 'sha256')
+    >>> hashfunc('sir robin').digest()
+
+    Args:
+        name (string): The string name of the desired algorithm
+        data (buffer): The buffer to hash
+    Returns:
+        callable: The callable hash function of the provided
+            algorithm updated with the data to be hashed
+    """
+    hashfunc = hashlib.new(name)
+    hashfunc.update(data)
+    return hashfunc
+
+
+def verify_signature(token, threshold=None):
+    """Verify the signature within an apple pay token according to
+    the #ApplePaySpec documentation.
+
+    Args:
+        token (dict): the PKPaymentToken object defined within the #ApplePaySpec
+    Kwargs:
+        threshold (timedelta, None): Amount of time to consider the token valid.
+            No validation will be performed if a treshold is not provided
```

Comment:
I think this means not validation of the signing time will be performed, right? We'll still do all the other steps.